### ISSUE TITLE
Patch for bootcode

### ISSUE THREAD

**Issue by pali** [dosfstools/dosfstools#72]
Sam Bingner in 2003 wrote patch for mkdosfs which adds support for writing custom boot code to FAT boot sector: https://lists.debian.org/debian-devel/2003/06/msg01316.html

This patch was included into Debian version 2.11-5 of mkdosfs, but reverted in 2.11-6 as it cause problems for syslinux. Daniel Baumann wrote that proper fix would be later. See bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=489292

Apparently Sam Bingner's patch was never fixed and never entered into upstream dosfstools.

It would be nice to have boot code support in mkfs.fat. I suspect that problem reported in above bug is because `bs.hidden` needs to be set to `bs.secs_track` only if using external boot code.

**Comment by luzemario** [comment]
So, including FreeDOS/SvarDOS bootcode is not an option?

**Comment by pali** [comment]
@luzemario: Has FreeDOS/SvarDOS patch for dosfstools's mkfs.fat which can add custom boot code to FAT boot sector?

**Comment by luzemario** [comment]
No. I mean to put FreeDOS/SvarDOS bootsector verbatim on the disk, and their IO.SYS/COMMAND.COM too, directly from dosfstools.As 15 years running Linux only, and having  hundreds of times the need for a single DOS boot disk without using pirated Windows only for making a boot disk, I would like to see a way to have working DOS boot disks created from Linux. I don't care if they come from another project, since their license allows this.Please get in touch with these projects if needed, and solve the DOS boot issue once and for all.
-------- Mensagem original --------De : pali ***@***.***> Data: 09/04/2022  07:13  (GMT-03:00) Para: dosfstools/dosfstools ***@***.***> Cc: Luzemário ***@***.***>, Mention ***@***.***> Assunto: Re: [dosfstools/dosfstools] Patch for bootcode (#72) 
@luzemario: Has FreeDOS/SvarDOS patch for dosfstools's mkfs.fat which can add custom boot code to FAT boot sector?

**Comment by pali** [comment]
> No. I mean to put FreeDOS/SvarDOS bootsector verbatim on the disk

And obviously this cannot be done due to this open issue.
